Added doc about Homestead's Symfony integration

wouterj opened the pull request:
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | all |
| Fixed tickets | - |

wouterj committed:
Added doc about Homestead's Symfony integration

javiereguiluz commented on cookbook/workflow/homestead.rst:
```diff
@@ -0,0 +1,66 @@
+
+Imagine you've installed your Symfony application in
+``~/projects/symfony_demo`` on your local system. You first need Homestead to
+sync your files in this project. Execute ``homestead edit`` to edit the
```
I find `sync` hard to understand. Is Homestead syncing my files with other folder (are they being copied/moved?), is Homestead just checking for their existance, etc.

WouterJ replied:
Homestead is syncing the local directory with the directory you mapped it to in the Homestead environment, I'll expand the description a bit.

javiereguiluz commented on cookbook/workflow/homestead.rst:
```diff
@@ -0,0 +1,66 @@
+          to: /home/vagrant/projects/symfony_demo/web
+          type: symfony
+
+The ``type`` option tells Homestead to use the Symfony Nginx configuration.
```
I think it's spelled `nginx` instead of `Nginx`.

WouterJ replied:
You're correct, Laravel docs confused me :)

javiereguiluz commented on cookbook/workflow/homestead.rst:
```diff
@@ -0,0 +1,66 @@
+.. seealso::
+
+    To learn more features of Homestead, including Blackfire Profiler
+    integration, automatic creation of MySQL databases and more, please read
```
The `please` word may be unnecessary here.

javiereguiluz commented on cookbook/workflow/homestead.rst:
```diff
@@ -0,0 +1,66 @@
+In order to develop a Symfony application, you might want to use a virtual
+development environment instead of the built-in server or WAMP/LAMP. Homestead_
+is an easy-to-use Vagrant_ box to get a virtual environment up and running
+quickly.
```
Would it be appropriate to expand this introduction explaining the advantages and drawbacks of using Homestead with Symfony? For example, this method may run Symfony significantly slower than the native method ... or maybe not, etc.

WouterJ replied:
I definitely think it's a good idea to link to http://www.whitewashing.de/2013/08/19/speedup_symfony2_on_vagrant_boxes.html in a tip box.

I'll also add another paragraph after this with the advantages/disadvantages of using vagrant.

javiereguiluz replied:
I'm worried about this because people usually brag about Vagrant advantages but forget its drawbacks. Besides being much much much slower than the native application, it causes lots of other minor issues/troubles/misunderstandings (like these [Symfony issues related (somehow) to Vagrant](https://github.com/symfony/symfony/issues?utf8=%E2%9C%93&q=is%3Aissue+vagrant+)).

I'm not saying that Vagrant is bad. But I'd like to state clear to readers that using Vagrant will make your Symfony run very slow and it may cause other (minor) troubles totally unrelated to Symfony.

And about linking to third-party resources, **I think I was wrong**, I apologize for that and I think we should start linking to trustworthy third-party resources, like Benjamin's blog.

WouterJ replied:
Many of the linked issues are related to someone saying "Hey, I tested this on a clean Vagrant box", which is exactly why Vagrant was created in the first place. I can't see many issues directly related to Vagrant bugs.

And following some guides from Benjamin's blog can speed up applications very nicely. In fact, on my Windows PC, applications run on Homestead are much faster and consume less CPU than those running on the built-in PHP webserver or Wamp.

WouterJ commented:
Updated this PR. I did not add a section about the advantages and drawbacks, I don't have enough knowledge to write these down. However, I added a quick tip to improve the speed of the application on Homestead. I would like to link to http://www.whitewashing.de/2013/08/19/speedup_symfony2_on_vagrant_boxes.html , but I guess that's not possible due to symfony.com's external linking policy (/cc @javiereguiluz)?

xabbuh commented on cookbook/workflow/homestead.rst:
```diff
@@ -0,0 +1,74 @@
+.. tip::
+
+    Due to the amount of filesystem operations in Symfony (e.g. updating cache
+    files and writing to log files), Symfony can slow dow signifcantly. To
```
slow dow**n**

xabbuh commented on cookbook/workflow/homestead.rst:
```diff
@@ -0,0 +1,74 @@
+        - map: ~/projects
+          to: /home/vagrant/projects
+
+The projects directory is now accessible at ``/home/vagrant/projects`` in the
```
Backticks around `projects`?

xabbuh commented on cookbook/workflow/homestead.rst:
```diff
@@ -0,0 +1,74 @@
+          to: /home/vagrant/projects/symfony_demo/web
+          type: symfony
+
+The ``type`` option tells Homestead to use the Symfony nginx configuration.
```
Nginx

WouterJ replied:
the official spelling is "nginx": http://nginx.org/

xabbuh replied:
oh, looks like we need to update some places then

WouterJ commented:
Updated

xabbuh commented:
Is this ready to be merged?

WouterJ commented:
Yes, (imo)

xabbuh commented:
@WouterJ Was is inteded to base this on the `2.8` branch?

WouterJ commented:
No, should work with all Symfony versions

xabbuh commented:
Thanks Wouter.
